File: 2108-Find_First_Palindromic_String_in_the_Array/Find_First_Palindromic_String_in_the_Array.go
package findfirstpalindromicstringinthearray

func firstPalindrome(words []string) string {
	for i := range words {
		if isPalindrome(words[i]) {
			return words[i]
		}
	}

	return ""
}

func isPalindrome(word string) bool {
	left := 0
	right := len(word) - 1

	for left < right {
		if word[left] != word[right] {
			return false
		}

		left++
		right--
	}

	return true
}

File: 2108-Find_First_Palindromic_String_in_the_Array/Find_First_Palindromic_String_in_the_Array.java
class Solution {
    public String firstPalindrome(String[] words) {
        for(int i = 0; i < words.length; i++){
            if(isPalindrome(words[i])) return words[i];
        }
        
        return "";
    }
    
    private boolean isPalindrome(String s){
        int left = 0;
        int right = s.length() - 1;
        
        while(left < right){
            if(s.charAt(left++) != s.charAt(right--)){
                return false;
            }
        }
        
        return true;
    }
}